core-bundle/src/Command/DatabaseImportCommand.php

```diff
@@ -0,0 +1,134 @@
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Imports.
+ *
+ * @internal
+ */
+class DatabaseImportCommand extends Command
+{
+    protected static $defaultName = 'contao:database:import';
+
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setAliases(['contao:db:import'])
+            ->addOption('truncate', 't', InputOption::VALUE_NONE, 'Truncate the existing database')
+            ->addArgument('file', InputArgument::REQUIRED, 'The path to the SQL dump.')
+            ->setDescription('Imports an SQL dump directly to the database.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if ($input->getOption('truncate')) {
+            try {
+                $this->truncate();
+            } catch (Exception $e) {
+                $io->error('Could not truncate database: '.$e->getMessage());
+
+                return 1;
+            }
+
+            $io->success('Successfully truncated existing data.');
+        }
+
+        $file = $input->getArgument('file');
+        $handle = strcasecmp(substr($file, -3), '.gz') ? fopen($file, 'r') : gzopen($file, 'rb');
+
+        if (!$handle) {
+            $io->error('SQL dump file not found.');
+
+            return 1;
+        }
+
+        try {
+            $this->import($handle);
+        } catch (Exception $e) {
+            $io->error('Could not import SQL dump: '.$e->getMessage());
+
+            return 1;
+        }
+
+        $io->success('Successfully imported SQL dump.');
+
+        return 0;
+    }
+
+    /**
+     * @param resource $handle
+     *
+     * @throws Exception
+     */
+    private function import($handle): void
+    {
+        $sql = '';
+        $delimiter = ';';
+
+        while ($s = fgets($handle)) {
+            if ('DELIMITER ' === strtoupper(substr($s, 0, 10))) {
+                $delimiter = trim(substr($s, 10));
+                continue;
+            }
+
+            if (substr($ts = rtrim($s), -\strlen($delimiter)) === $delimiter) {
+                $sql .= substr($ts, 0, -\strlen($delimiter));
+                $this->connection->executeQuery($sql);
+                $sql = '';
+                continue;
+            }
+
+            $sql .= $s;
+        }
+
+        if ('' !== rtrim($sql)) {
+            $this->connection->executeQuery($sql);
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function truncate(): void
+    {
+        $tables = $this->connection->getSchemaManager()->listTableNames();
+
+        foreach ($tables as $table) {
+            if (0 === strncmp($table, 'tl_', 3)) {
```

Comment:
This is copied from the current installtaion bundle. Not sure we should keep this?

Reply:
The prefix doesn't make much sense to me. But I wonder if we should rather drop the whole truncation logic and let the SQL files handle that themselves? They can include the schema and arbitrary commands anyway.

Maybe we could pair the command with a `contao:database:dump` command that creates compatible files?

Reply:
Yes I will provide one but it's not related.